src/System.Buffers/tests/BasicUnitTests.cs

```diff
@@ -40,5 +40,57 @@ public void ByteSpanEmptyCreateArrayTest()
             var array = empty.CreateArray();
             Assert.Equal(0, array.Length);
         }
+        
+        [Fact]
+        public unsafe void ByteSpanEqualsTestsTwoDifferentBuffersSameValues()
+        {
+            byte[] buffer1 = new byte[128];
+            byte[] buffer2 = new byte[128];
+            
+            for (int i = 0; i < 128; i++)
+            {
+                buffer1[i] = (byte)(129 - i);
+                buffer2[i] = (byte)(129 - i);
+            }
+            
+            fixed (byte* buffer1pinned = buffer1) fixed (byte* buffer2pinned = buffer2)
```

Comment:
Nice one, I actually didn't know you can write it in the same statement. Will change it to one of those.